### PR TITLE
Remove `jna-posix`

### PR DESCRIPTION
**Pull request by basil**
`jna-posix` was superseded by `jnr-posix` in Jenkins core in 2013 in commit 2a1d1631bd, but our fork of `org.jruby.ext.posix:jna-posix` (1.0.3-jenkins-1) remains present for backward compatibility. `jna-posix` is not compatible with Java 10+. Carrying this dependency seems undesirable for the Jenkins project in 2021. Isn't it time we moved on?

This PR makes incompatible changes to `PosixException` to remove the exposure of `jna-posix`. I used `usage-in-plugins` to look for usages of `PosixException`. `PosixException` is used in [Copy Artifact](https://plugins.jenkins.io/copyartifact/), [CloudBees Backup Plugin](https://docs.cloudbees.com/docs/release-notes/latest/plugins/infradna-backup-plugin/), and [Operations Center Context Plugin](https://docs.cloudbees.com/docs/release-notes/latest/plugins/operations-center-context-plugin/). But in none of these three cases is the constructor or the `getErrorCode()` method called. They are all just catching `PosixException`. So removing the second argument from the constructor and removing `getErrorCode()` shouldn't break anything.

I also used `usage-in-plugins` to look for usages of classes in the `jna-posix` JAR as well as `hudson.os.PosixAPI#get`, `hudson.os.PosixAPI#isNative`, and `hudson.os.PosixAPI#supportsNative`. Here we go again. `hudson/os/PosixAPI#get()Lorg/jruby/ext/posix/POSIX;` is still used in three ancient plugins:

- [Maven Repository Scheduled Cleanup](https://plugins.jenkins.io/maven-repo-cleaner/), a plugin last released 8 years ago with 3,749 installations. This dependency was already removed in jenkinsci/maven-repo-cleaner-plugin#4, but this change remains unreleased three years later.
- [SICCI for Xcode](https://plugins.jenkins.io/sicci_for_xcode/), a plugin last released 10 years ago with 519 installations. I opened jenkinsci/tmpcleaner-plugin#7 to migrate this plugin from `jna-posix` to `jnr-posix`.
- [java.io.tmpdir cleaner](https://plugins.jenkins.io/tmpcleaner/), a plugin last released 5 years ago with 884 installations. I opened jenkinsci/sicci-for-xcode-plugin#2 to migrate this plugin from `jna-posix` to `jnr-posix`.

I propose we move forward with this removal whether these plugins get released or not. `jna-posix` has been deprecated in Jenkins core for 8 years, and it won't work with Java 11. It's time to move on. Once this PR is merged, I'll file a PR in the Update Center to officially mark these three plugins as deprecated.

### Proposed changelog entries

Developer: Remove `jna-posix` dependency from Jenkins core. Plugins that use `jna-posix` functionality, including [Maven Repository Scheduled Cleanup](https://plugins.jenkins.io/maven-repo-cleaner/), [SICCI for Xcode](https://plugins.jenkins.io/sicci_for_xcode/), and [java.io.tmpdir cleaner](https://plugins.jenkins.io/tmpcleaner/), must be migrated from `jna-posix` to `jnr-posix`.

### Proposed upgrade guidelines

The `jna-posix` dependency has been removed from Jenkins core. Plugins that use `jna-posix` functionality, including [Maven Repository Scheduled Cleanup](https://plugins.jenkins.io/maven-repo-cleaner/), [SICCI for Xcode](https://plugins.jenkins.io/sicci_for_xcode/), and [java.io.tmpdir cleaner](https://plugins.jenkins.io/tmpcleaner/), are no longer supported and must be removed.

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Review by jglick — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
